cloud/filename.h

```diff
@@ -150,10 +151,10 @@ inline bool IsWalFile(const std::string& pathname) {
 
 inline bool IsManifestFile(const std::string& pathname) {
   // extract last component of the path
-  std::string fname;
+  folly::StringPiece fname;
```

Comment:
I don't think rocksdb have folly as dependency. You can use `rocksdb::Slice` and there is a `starts_with` function there

Reply:
<img width="247" alt="Screenshot 2023-07-10 at 3 41 05 PM" src="https://github.com/rockset/rocksdb-cloud/assets/3534188/879b1c55-1d1a-4dbd-8d6c-bdaee7e1fff5">

looks like there are a bunch of reference to folly

Reply:
you need to compile with `USE_FOLLY` to enable that. I barely remember the rocksdb-cloud tests don't work when compiling with `USE_FOLLY`, so you have to provide an option when `USE_FOLLY` is disabled and fix the issue of rocksdb-cloud tests.

Reply:
so, one last question hopefully, MANIFEST can have postfix but not prefix right? cause it used to be "find(manifest)", now it is "starts_with(manifest)"